models/oci/oci.go

```diff
@@ -112,3 +120,105 @@ func getLayerMediaType(extension string) types.MediaType {
 	}
 	return types.MediaType(fmt.Sprintf("%s.%s", oci.CanonicalMediaTypePrefix, extension))
 }
+
+// function to pull models from any OCI-compatible repository
+func PushToOCIRegistry(dirPath, registryAdd, repositoryAdd, imageTag, username, password string) error {
+
+	// Create a new file store
+	fs, err := file.New(".")
+	if err != nil {
+		return ErrFileNotFound(err)
```

Comment:
file.New creates a file, I believe.

Reply:
We're creating a new file to package the manifest. With each iteration, the information from the previous file will be overwritten, and the file info will be continuously appended into the descriptors.

Reply:
So, if an error occurs you should return `ErrWriteFile` / `ErrWriteLayer`.

Reply:
That makes sense.